rustworkx/__init__.pyi

```diff
@@ -213,7 +222,281 @@ from .token_swapper import graph_token_swapper as graph_token_swapper
 from .union import digraph_union as digraph_union
 from .union import graph_union as graph_union
 
-S = TypeVar("S")
-T = TypeVar("T")
+_S = TypeVar("_S")
+_T = TypeVar("_T")
+_BFSVisitor = TypeVar("_BFSVisitor", bound=visit.BFSVisitor)
+_DFSVisitor = TypeVar("_DFSVisitor", bound=visit.DFSVisitor)
+_DijkstraVisitor = TypeVar("_DijkstraVisitor", bound=visit.DijkstraVisitor)
+
+class PyDAG(Generic[_S, _T], PyDiGraph[_S, _T]): ...
 
-class PyDAG(Generic[S, T], PyDiGraph[S, T]): ...
+def distance_matrix(
+    graph: PyGraph | PyDiGraph,
+    parallel_threshold: int = ...,
+    as_undirected: bool = ...,
+    null_value: float = ...,
+) -> np.ndarray: ...
+def unweighted_average_shortest_path_length(
+    graph: PyGraph | PyDiGraph,
+    parallel_threshold: int = ...,
+    disconnected: bool = ...,
+) -> float: ...
+def adjacency_matrix(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    weight_fn: Callable[[_T], float] | None = ...,
+    default_weight: float = ...,
+    null_value: float = ...,
+) -> np.ndarray: ...
+def all_simple_paths(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    from_: int,
+    to: int,
+    min_depth: int | None = ...,
+    cutoff: int | None = ...,
+) -> list[list[int]]: ...
+def floyd_warshall(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    weight_fn: Callable[[_T], float] | None = ...,
+    default_weight: float = ...,
+    parallel_threshold: int = ...,
+) -> AllPairsPathLengthMapping: ...
+def floyd_warshall_numpy(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    weight_fn: Callable[[_T], float] | None = ...,
+    default_weight: float = ...,
+    parallel_threshold: int = ...,
+) -> np.ndarray: ...
+def floyd_warshall_successor_and_distance(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    /,
```

Comment:
Do we actually enforce this in the universal function? The python function signature doesn't say that `graph` is positional only.

```suggestion
```

Reply:
Done